yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,334 @@
+"""
+Chimera data structures
+
+
+
+"""
+
+# -----------------------------------------------------------------------------
+# Copyright (c) 2013, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# -----------------------------------------------------------------------------
+
+import os
+
+import numpy as np
+
+from yt.geometry.geometry_handler import YTDataChunk
+from yt.geometry.unstructured_mesh_handler import UnstructuredIndex
+from yt.utilities.file_handler import HDF5FileHandler
+from yt.utilities.io_handler import io_registry
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+try:
+    from yt.data_objects.unstructured_mesh import SemiStructuredMesh
+except ModuleNotFoundError:
+    from yt.data_objects.index_subobjects.unstructured_mesh import SemiStructuredMesh
+
+
+from yt.data_objects.static_output import Dataset
+
+from .fields import ChimeraFieldInfo
+
+
+class ChimeraMesh(SemiStructuredMesh):
+
+    _index_offset = 0
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+def _find_files(
+    filename_c,
+):  # Returns a list of all files that share a frame number with the input
+    dirname, file = os.path.split(filename_c)
+    dir_contents = os.listdir(dirname)
+    frames = [f for f in dir_contents if file[:-7] in f]
+    directory = os.path.dirname(filename_c)
+    index_filenames = [directory + "/" + f for f in frames]
+    index_filenames.sort()
+    return index_filenames
+
+
+class ChimeraUNSIndex(UnstructuredIndex):
+    def __init__(self, ds, dataset_type="chimera"):
+        self._handle = ds._handle
+        super().__init__(ds, dataset_type)
+        self.directory = os.path.dirname(self.dataset.filename)
+        self.dataset_type = dataset_type
+
+    def _initialize_mesh(self):
+        self.meshes = []
+        index_filenames = _find_files(
+            self.dataset.filename
+        )  # Retrieves list of all datafiles with the same frame number
+        if any(
+            "grid_2" in file for file in index_filenames
+        ):  # Detects Yin-Yang data format
+            yy = True
+        else:
+            yy = False
```

Comment:
This is equivalent to
```suggestion
        # Detects Yin-Yang data format
        yy = any("grid_2" in file for file in index_filenames)
```

Reply:
Replaced in 10df6b4